xbmc/cores/VideoPlayer/Edl.cpp

```diff
@@ -770,19 +770,19 @@ int CEdl::RemoveCutTime(int iSeek) const
   return iSeek - iCutTime;
 }
 
-int CEdl::RestoreCutTime(int iClock) const
+double CEdl::RestoreCutTime(double dClock) const
 {
   if (!HasCut())
-    return iClock;
+    return dClock;
 
-  int iSeek = iClock;
+  double dSeek = dClock;
   for (int i = 0; i < (int)m_vecCuts.size(); i++)
   {
-    if (m_vecCuts[i].action == CUT && iSeek >= m_vecCuts[i].start)
-      iSeek += m_vecCuts[i].end - m_vecCuts[i].start;
+    if (m_vecCuts[i].action == CUT && dSeek >= m_vecCuts[i].start)
+      dSeek += (double)m_vecCuts[i].end - (double)m_vecCuts[i].start;
```

Comment:
I removed the cast. If I understand C++ correctly, the calculation is done in double any way as it expands to dSeek = dSeek + int.
As we calculate here with a double, the int is promoted to a double.
Can you confirm that?